### PR TITLE
[multi-asic]: remove load_sonic_global_db_config calls

### PR DESCRIPTION
**Pull request by arlakshm**
Signed-off-by: Arvindsrinivasan Lakshmi Narasimhan <arlakshm@microsoft.com>

#### Why I did it
Remove the call to `SonicDBConfig.load_sonic_global_db_config()` in the multi asic functions.
The expection is the client calling this function will call `SonicDBConfig.load_sonic_global_db_config()`

This PR is dependent on the PR https://github.com/Azure/sonic-utilities/pull/1712 
#### How to verify it
compile sonic-utilities

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: src/sonic-py-common/sonic_py_common/multi_asic.py [hunk 1]
```diff
@@ -456,3 +452,14 @@ def validate_namespace(namespace):
         return True
     else:
         return False
+
+
+
+def load_db_config():
```
Comment (qiluo-msft) [inline src/sonic-py-common/sonic_py_common/multi_asic.py:458]: Use 1 or 2 empty lines between functions.

File: src/sonic-host-services/scripts/caclmgrd [hunk 2]
```diff
@@ -109,7 +109,7 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
         self.namespace_docker_mgmt_ip = {}
         self.namespace_docker_mgmt_ipv6 = {}
 
-        namespaces = device_info.get_all_namespaces()
```
Comment (qiluo-msft) [inline src/sonic-host-services/scripts/caclmgrd:112]: >device_info.get_all_namespaces [](http://example.com/codeflow?start=21&length=30)

Will you remove the function definition completely?

**Comment by qiluo-msft** [comment]
>             swsscommon.SonicDBConfig.initializeGlobalConfig()

Is it same as load_db_config()? If yes, we can call that function, and remove device_info from this file.

---
Refers to: src/sonic-host-services/scripts/caclmgrd:617 in bfa16f9. [](commit_id = bfa16f9b04725011f5d697a317f4834b72c5b3eb, deletion_comment = False)

**Comment by qiluo-msft** [comment]
The vstest failure is relating this PR
```
teamd/test_portchannel.py:17: AssertionError
----------------------------- Captured stdout call -----------------------------
-----rc=1 for cmd config portchannel add PortChannel0001-----
Traceback (most recent call last):
  File "/usr/local/bin/config", line 8, in <module>
    sys.exit(config())
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 1134, in invoke
    Command.invoke(self, ctx)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/config/main.py", line 954, in config
    SonicDBConfig.initialize()
RuntimeError: SonicDBConfig already initialized
```

File: src/sonic-py-common/sonic_py_common/multi_asic.py [hunk 3]
```diff
@@ -456,3 +452,14 @@ def validate_namespace(namespace):
...
+    '''
+    Load the database_global.json for multi asic
+    '''
+    if is_multi_asic():
+        swsscommon.SonicDBConfig.load_sonic_global_db_config()
+    else:
+        swsscommon.SonicDBConfig.initialize()
```
Comment (qiluo-msft) [inline src/sonic-py-common/sonic_py_common/multi_asic.py:465]: Not a blocking issue. You may use `load_sonic_db_config` which is available in both swsscommon and swsssdk.

File: src/sonic-host-services/scripts/caclmgrd (hunk 2 above)
Comment (arlakshm) [reply src/sonic-host-services/scripts/caclmgrd:112]: Yes, for all the multi asic specific functions we need to use the ones defined in multi_asic.py. 
The function in device_info.py will be removed

File: src/sonic-py-common/sonic_py_common/multi_asic.py (hunk 1 above)
Comment (arlakshm) [reply src/sonic-py-common/sonic_py_common/multi_asic.py]: This function is moved to sonic_utilities in the PR https://github.com/Azure/sonic-utilities/pull/1712

**Comment by arlakshm** [comment]
> The vstest failure is relating this PR
> 
> ```
> teamd/test_portchannel.py:17: AssertionError
> ----------------------------- Captured stdout call -----------------------------
> -----rc=1 for cmd config portchannel add PortChannel0001-----
> Traceback (most recent call last):
>   File "/usr/local/bin/config", line 8, in <module>
>     sys.exit(config())
>   File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 764, in __call__
>     return self.main(*args, **kwargs)
>   File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 717, in main
>     rv = self.invoke(ctx)
>   File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 1134, in invoke
>     Command.invoke(self, ctx)
>   File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 956, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 555, in invoke
>     return callback(*args, **kwargs)
>   File "/usr/local/lib/python3.7/dist-packages/click/decorators.py", line 17, in new_func
>     return f(get_current_context(), *args, **kwargs)
>   File "/usr/local/lib/python3.7/dist-packages/config/main.py", line 954, in config
>     SonicDBConfig.initialize()
> RuntimeError: SonicDBConfig already initialized
> ```

Hi Qi,
As discussed this error is caused because  of the following sequence:

-	For every CLI the file utilities_common.cli is imported. In this file we initialize the global variable
`iface_alias_converter = InterfaceAliasConverter()`
-	In the __init__ of the class InterfaceAliasConverter we call the` connect_config_db_for_ns() `with namespace option
For single asic the namespace is default_namespace(“ “)
-	In the __init__ of the sonicV2connector class we call the function `get_db_list(). `
-	In this function if the db is not initialized we initialize the db.
Thereby loading the database file
-	So in `main.py` when we try to initialize the db again it errors out

To prevent this error, added a new function `load_db_config` in sonic-utilities to safely load the database config files.
the  changes are in PR https://github.com/Azure/sonic-utilities/pull/1712.

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**